### PR TITLE
Add the instruction on how to upgrade Knative with the operator

### PR DESCRIPTION
**Pull request by houshengbo**
## Proposed Changes 

- This PR adds the instruction on how to upgrade Knative with the operator.

### CODE REVIEW

File: docs/install/upgrade-installation-with-operator.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+metadata:
+  name: knative-eventing
+  namespace: knative-eventing
+spec:
+  version: "0.20"
+```
+
+## Verifying the upgrade
+
+To confirm that your components and plugins have successfully upgraded, view the status of their pods in the relevant namespaces.
```
Comment (matzew) [inline docs/install/upgrade-installation-with-operator.md:114]: `plugins` ?

File: docs/install/upgrade-installation-with-operator.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+v0.18 before attempting to upgrade to v0.19. The attribute `spec.version` is the only field you need to change in the
+Serving or Eventing CR to perform an upgrade. You do not need to specify the version in terms of the `patch` number,
+because the Knative Operator will match the latest available `patch` number, as long as you specify `major.minor` for
+the version. For example, you only need to specify `0.19` to upgrade to the latest v0.19 release. There is no need to
+know the exact `patch` number.
+
+The Knative Operator implements a minus 3 principle to support the Knative versions, which means the current version
+of the Operator can support Knative with the version back 3 in terms of the `minor` number. For example, if the
+current version of the Operator is 0.19.x, it bundles and supports the installation of Knative with the versions,
+0.16.x, 0.17.x, 0.18.x and 0.19.x.
```
Comment (matzew) [inline docs/install/upgrade-installation-with-operator.md:18]: Can I upgrade directly from 0.17 ... to 0.19 ? 

If no: let's add a note here

Comment (houshengbo) [reply docs/install/upgrade-installation-with-operator.md:18]: Check line 7: 
```
The upgrade is relatively more straightforward, if you install Knative with the Knative Operator. We support upgrading by a single [minor](https://semver.org/) version number. For example, if you have v0.17 installed, you must upgrade to v0.18 before attempting to upgrade to v0.19.
```

**Review by evankanderson — COMMENTED** [review]
Just a few comments on the introductory text, thanks for writing this up and consolidating it!

**Pull request merged**